server/src/main/java/org/apache/druid/client/ServerViewUtil.java

```diff
@@ -42,17 +44,21 @@ public static List<LocatedSegmentDescriptor> getTargetLocations(
       TimelineServerView serverView,
       String datasource,
       List<Interval> intervals,
-      int numCandidates
+      int numCandidates,
+      HistoricalFilter historicalFilter,
+      CloneQueryMode cloneQueryMode
   )
   {
-    return getTargetLocations(serverView, new TableDataSource(datasource), intervals, numCandidates);
+    return getTargetLocations(serverView, new TableDataSource(datasource), intervals, numCandidates, historicalFilter, cloneQueryMode);
```

Comment:
this also would only need `cloneQueryMode` if the config was already on `ServerSelector`